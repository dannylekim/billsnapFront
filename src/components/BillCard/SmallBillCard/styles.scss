@import '../../../variables';

div#active__bill {
  background-color: rgba(241, 241, 241, 0.71);
}

div.bill__items {
  text-align: left;

  .card-text {
    width: 100%;
    margin-top: 25px;
    height: 50px;
    display: flex;

    .bill__name__price {
      width: 100%;

      #bill__name {
        margin-left: 10px;
        font-family: $primary-font;
        font-style: normal;
        font-weight: bold;
        float: left;
        color: $primary-text-color;
      }

      #bill__balance {
        float: right;
      }
    }
  }

  .split__by {
    span#responsible {
      margin-left: 32px;
      font-weight: bold;
      font-family: $primary-font;
      color: $tertiary-text-color;
    }
  }

  .card-footer {
    width: 100%;
    margin-left: 25px;
    background-color: $card-color;
    padding: 0;
  }
}

#bill__balance {
  font-family: $primary-font;
  font-style: normal;
  font-weight: bold;
  font-size: 16px;
  color: $primary-price-color;
}