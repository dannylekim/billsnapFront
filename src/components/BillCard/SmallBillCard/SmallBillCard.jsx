import React from "react";
import PropType from "prop-types";
import {FaBus, FaCar, FaQuestion, FaShoppingBag, FaShoppingCart, FaUtensils } from "react-icons/fa";

import { filterDateTime } from '../../../helpers/DateTime';

import "./styles.scss";

const getBillIcons = (category) => {
  const color = "rgba(0, 0, 0, 0.96)";
  switch (category) {
    case "food":
      return <FaUtensils color={color} size={24} />;
    case "transport":
      return <FaCar color={color} size={24} />;
    case "public-transport":
      return <FaBus color={color} size={24} />;
    case "grocery":
      return <FaShoppingCart color={color} size={24} />;
    case "shopping":
      return <FaShoppingBag color={color} size={24} />;
    default:
      return <FaQuestion color={color} size={24} />;
  }
};

const SmallBillCard = ({ activeBill, bill }) => (
  <div
    className="bill__items card-body"
    id={
      activeBill && activeBill.id === bill.id
        ? "active__bill"
        : "non__active__bill"
    }
  >
    {!!bill.created && (
      <div className="bill__items card-title text-muted">
        <span className="float-right" id="bill__created">
          {" "}
          {filterDateTime(bill.created)}
        </span>
      </div>
    )}
    <div className="bill__items card-text">
      <div id="bill__icon">{getBillIcons(bill.category)}</div>
      <div className="bill__name__price">
        <span id="bill__name">{bill.name} </span>
        <span id="bill__balance">
          {" "}
          {parseFloat(bill.balance).toFixed(2)} ${" "}
        </span>
      </div>
    </div>

    {bill.responsible && (
      <div className="bill__items split__by">
        <span id="responsible"> Split by : {bill.responsible.firstName} </span>
      </div>
    )}
  </div>
);

SmallBillCard.propTypes = {
  bill: PropType.arrayOf(PropType.shape({})), // array of bill object
  activeBill: PropType.shape({}) // bill selected
}

export default SmallBillCard;
