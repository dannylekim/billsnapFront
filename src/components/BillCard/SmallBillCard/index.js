import SmallBillCard from './SmallBillCard.jsx';
export default SmallBillCard;