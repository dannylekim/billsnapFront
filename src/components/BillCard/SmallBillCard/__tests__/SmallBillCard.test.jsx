import React from "react";
import SmallBillCard from "../SmallBillCard.jsx";

const fakeBills = [
  {
    id: 48,
    name: "Mcdonald",
    status: "OPEN",
    category: "food",
    created: "28-07-2020 14:25:10 -0400",
    balance: 253.89,
    responsible: {
      firstName: 'Hello',
      lastName: 'Motto'
    }
  }
];

describe("SmallBillCard", () => {
  describe("render", () => {
    describe("snapshots ðŸ“¸", () => {
      it("SmallBillCard should match snap shot", () => {
        matches(
          <SmallBillCard
            bill={fakeBills[0]}
            activeBillId={999}
          />
        );
      });

      it("SmallBillCard should match snap shot if owner exists", () => {
        matches(
          <SmallBillCard
            bill={fakeBills[0]}
            activeBillId={fakeBills[0].id}
          />
        );
      });
    });
  });
});
