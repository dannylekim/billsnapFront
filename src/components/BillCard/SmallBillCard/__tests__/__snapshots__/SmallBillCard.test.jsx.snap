// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallBillCard render snapshots ðŸ“¸ SmallBillCard should match snap shot 1`] = `
<div
  className="bill__items card-body bill-active"
>
  <div
    className="bill__items card-title text-muted"
  >
    <span
      className="float-right"
      id="bill__created"
    >
       
      28/07
    </span>
  </div>
  <div
    className="bill__items card-text"
  >
    <div
      id="bill__icon"
    >
      <svg
        color="rgba(0, 0, 0, 0.96)"
        fill="currentColor"
        height={24}
        size={24}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.96)",
          }
        }
        viewBox="0 0 416 512"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M207.9 15.2c.8 4.7 16.1 94.5 16.1 128.8 0 52.3-27.8 89.6-68.9 104.6L168 486.7c.7 13.7-10.2 25.3-24 25.3H80c-13.7 0-24.7-11.5-24-25.3l12.9-238.1C27.7 233.6 0 196.2 0 144 0 109.6 15.3 19.9 16.1 15.2 19.3-5.1 61.4-5.4 64 16.3v141.2c1.3 3.4 15.1 3.2 16 0 1.4-25.3 7.9-139.2 8-141.8 3.3-20.8 44.7-20.8 47.9 0 .2 2.7 6.6 116.5 8 141.8.9 3.2 14.8 3.4 16 0V16.3c2.6-21.6 44.8-21.4 48-1.1zm119.2 285.7l-15 185.1c-1.2 14 9.9 26 23.9 26h56c13.3 0 24-10.7 24-24V24c0-13.2-10.7-24-24-24-82.5 0-221.4 178.5-64.9 300.9z"
        />
      </svg>
    </div>
    <div
      className="bill__name__price"
    >
      <span
        id="bill__name"
      >
        Mcdonald
         
      </span>
      <span
        id="bill__balance"
      >
         
        253.89
         $
         
      </span>
    </div>
  </div>
</div>
`;

exports[`SmallBillCard render snapshots ðŸ“¸ SmallBillCard should match snap shot if owner exists 1`] = `
<div
  className="bill__items card-body false"
>
  <div
    className="bill__items card-title text-muted"
  >
    <span
      className="float-right"
      id="bill__created"
    >
       
      02/07
    </span>
  </div>
  <div
    className="bill__items card-text"
  >
    <div
      id="bill__icon"
    >
      <svg
        color="rgba(0, 0, 0, 0.96)"
        fill="currentColor"
        height={24}
        size={24}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.96)",
          }
        }
        viewBox="0 0 512 512"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M488 128h-8V80c0-44.8-99.2-80-224-80S32 35.2 32 80v48h-8c-13.25 0-24 10.74-24 24v80c0 13.25 10.75 24 24 24h8v160c0 17.67 14.33 32 32 32v32c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32v-32h192v32c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32v-32h6.4c16 0 25.6-12.8 25.6-25.6V256h8c13.25 0 24-10.75 24-24v-80c0-13.26-10.75-24-24-24zM112 400c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32zm16-112c-17.67 0-32-14.33-32-32V128c0-17.67 14.33-32 32-32h256c17.67 0 32 14.33 32 32v128c0 17.67-14.33 32-32 32H128zm272 112c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32z"
        />
      </svg>
    </div>
    <div
      className="bill__name__price"
    >
      <span
        id="bill__name"
      >
        KFC
         
      </span>
      <span
        id="bill__balance"
      >
         
        151.00
         $
         
      </span>
    </div>
  </div>
  <div
    className="bill__items split__by"
  >
    <span
      id="responsible"
    >
       Split by : 
      Hello
       
    </span>
  </div>
</div>
`;
